numpy/lib/function_base.py

```diff
@@ -1897,24 +1899,25 @@ def diff(a, n=1, axis=-1):
     if n < 0:
         raise ValueError(
             "order must be non-negative but got " + repr(n))
+
     a = asanyarray(a)
     nd = a.ndim
-    slice1 = [slice(None)]*nd
-    slice2 = [slice(None)]*nd
-    slice1[axis] = slice(1, None)
-    slice2[axis] = slice(None, -1)
-    slice1 = tuple(slice1)
-    slice2 = tuple(slice2)
+    rng = range(nd)
+    axis = rng[axis]
+    keep = slice(None)
 
-    if a.dtype == np.bool_:
-        da = a[slice1] != a[slice2]
-    else:
-        da = a[slice1] - a[slice2]
+    if n >= a.shape[axis]:
+        slice1 = tuple(slice(0, 0) if i == axis else keep for i in rng)
+        return a[slice1]
```

Comment:
I will add a test to make sure that the time-delta case works out as expected. If it lets me, I may retain the optimization, but with `>` instead of `>=`.